Mathlib/NumberTheory/ModularForms/EisensteinSeries/UniformConvergence.lean

```diff
@@ -0,0 +1,364 @@
+    add_zero, normSq_apply, add_re, mul_re, sub_zero, add_im] at *
+  exact H1
+
+theorem auxbound2 (z : ℍ) (δ ε : ℝ) (hε : 1 ≤ ε^2) : r z ≤ Complex.abs (δ * (z : ℂ) + ε) := by
+  rw [r, Complex.abs, min_le_iff]
+  have H1 : Real.sqrt (r1 z) ≤ Real.sqrt ((δ * (z : ℂ).re + ε) * (δ * (z : ℂ).re + ε ) +
+      δ * (z : ℂ).im * (δ * (z : ℂ).im)) := by
+    rw [r1, Real.sqrt_le_sqrt_iff, ← pow_two, ← pow_two]
+    apply r1_bound z δ ε hε
+    nlinarith
+  right
+  simp only [ne_eq, coe_re, coe_im, normSq_apply, AbsoluteValue.coe_mk, MulHom.coe_mk, add_re,
+    mul_re, ofReal_re, ofReal_im, zero_mul, sub_zero, int_cast_re, add_im, mul_im, add_zero,
+    int_cast_im] at *
+  exact H1
+
+--any suggestions as to where to put the lemmas below?
+lemma one_le_sq_div_abs_sq (a : ℝ) (ha : a ≠ 0) : 1 ≤ a^2 / |a|^2 := by
+    rw [_root_.sq_abs, le_div_iff']
+    simp only [mul_one, le_refl]
+    exact (sq_pos_iff a).mpr ha
+
+lemma int_abs_eq_complex_abs (a : ℤ) : Complex.abs a = a.natAbs := by
+  have : Complex.abs a = Complex.abs (a : ℝ) := by rfl
+  rw [this, abs_ofReal]
+  norm_cast
+  rw [Int.abs_eq_natAbs a]
+  rfl
+
+lemma ne_zero_if_max (x : Fin 2 → ℤ) (hx : x ≠ 0)
+    (h : (max (x 0).natAbs (x 1).natAbs) = (x 0).natAbs) : (x 0) ≠ 0 := by
+  intro h0
+  rw [h0] at h
+  simp only [ne_eq, Int.natAbs_zero, ge_iff_le, zero_le, max_eq_right, Int.natAbs_eq_zero] at *
+  have : x = ![x 0, x 1] := List.ofFn_inj.mp rfl
+  rw [h0, h] at this
+  simp only [this, Matrix.cons_eq_zero_iff, Matrix.zero_empty, and_self, not_true_eq_false] at hx
+
+lemma ne_zero_if_max' (x : Fin 2 → ℤ) (hx : x ≠ 0)
+    (h : (max (x 0).natAbs (x 1).natAbs) = (x 1).natAbs) : (x 1) ≠ 0 := by
+  apply ne_zero_if_max ![x 1, x 0] ?_ (by simpa using h)
+  simp only [ne_eq, Matrix.cons_eq_zero_iff, Matrix.zero_empty, and_true, not_and]
+  intro h1 h0
+  rw [fun_ne_zero_cases, h1, h0] at hx
+  simp only [ne_eq, not_true_eq_false, or_self] at *
+
+lemma div_max_sq_ge_one (x : Fin 2 → ℤ) (hx : x ≠ 0) :
+    (1 : ℝ) ≤ (x 0 / (max (x 0).natAbs (x 1).natAbs))^2 ∨
+      (1 : ℝ) ≤ (x 1 / (max (x 0).natAbs (x 1).natAbs))^2 := by
+  cases' (max_choice (x 0).natAbs (x 1).natAbs) with H1 H2
+  · left
+    rw [H1]
+    have : (x 0 : ℝ) ≠ 0 := by
+      simpa using (ne_zero_if_max x hx H1)
+    have h1 := one_le_sq_div_abs_sq (x 0 : ℝ) this
+    simp only [ne_eq, max_eq_left_iff, Int.cast_eq_zero, int_cast_abs, div_pow, ge_iff_le] at *
+    convert h1
+    norm_cast
+    rw [int_abs_eq_complex_abs]
+  · right
+    rw [H2]
+    have : (x 1 : ℝ) ≠ 0 := by
+      simpa using (ne_zero_if_max' x hx H2)
+    have h1 := one_le_sq_div_abs_sq (x 1 : ℝ) this
+    simp only [ne_eq, max_eq_right_iff, Int.cast_eq_zero, int_cast_abs, div_pow, ge_iff_le] at *
+    convert h1
+    norm_cast
+    rw [int_abs_eq_complex_abs]
+
+/-This should work for complex `k` (with a condition on `k.re`) but last I checked we were missing
+some lemmas about this -/
```

Comment:
Wel... let's add what is missing if it is not too much work!

Reply:
done!